public/app/plugins/datasource/tempo/QueryEditor/NativeSearch.tsx

```diff
@@ -68,72 +63,59 @@ const NativeSearch = ({ datasource, query, onChange, onBlur, onRunQuery }: Props
     spanName: false,
   });
 
-  async function fetchOptionsCallback(name: string, lp: TempoLanguageProvider, query = '') {
-    try {
-      setIsLoading((prevValue) => ({ ...prevValue, [name]: false }));
-      const options = await lp.getOptions(name);
-      const filteredOptions = options.filter((item) => (item.value ? fuzzyMatch(item.value, query).found : false));
-      return filteredOptions;
-    } catch (error) {
-      if (error?.status === 404) {
-        setIsLoading((prevValue) => ({ ...prevValue, [name]: false }));
-      } else {
-        dispatch(notifyApp(createErrorNotification('Error', error)));
+  const loadOptions = useCallback(
+    async (name: string, query = '') => {
+      const lpName = name === 'serviceName' ? 'service.name' : 'name';
+      setIsLoading((prevValue) => ({ ...prevValue, [name]: true }));
+
+      try {
+        const options = await languageProvider.getOptions(lpName);
+        const filteredOptions = options.filter((item) => (item.value ? fuzzyMatch(item.value, query).found : false));
+        return filteredOptions;
+      } catch (error) {
+        if (error?.status === 404) {
+          setError(error);
+        } else {
+          dispatch(notifyApp(createErrorNotification('Error', error)));
+        }
+        return [];
+      } finally {
         setIsLoading((prevValue) => ({ ...prevValue, [name]: false }));
       }
-      setError(error);
-      return [];
-    }
-  }
-
-  const loadOptionsOfType = useCallback(
-    (name: string) => {
-      setIsLoading((prevValue) => ({ ...prevValue, [name]: true }));
-      return fetchOptionsCallback(name, languageProvider);
     },
     [languageProvider]
   );
 
-  const fetchOptionsOfType = useCallback(
-    (name: string) => debounce(() => loadOptionsOfType(name), 500, { leading: true, trailing: true }),
-    [loadOptionsOfType]
-  );
-
-  const getNameOptions = (query: string, name: string) => {
-    setIsLoading((prevValue) => ({ ...prevValue, [name]: true }));
-    return fetchOptionsCallback(name, languageProvider, query);
-  };
-
-  const getServiceNameOptions = (query: string) => {
-    return getNameOptions(query, 'service.name');
-  };
-
-  const getSpanNameOptions = (query: string) => {
-    return getNameOptions(query, 'name');
-  };
-
-  const serviceNameSearch = debounce(getServiceNameOptions, 500, { leading: true, trailing: true });
-  const spanNameSearch = debounce(getSpanNameOptions, 500, { leading: true, trailing: true });
-
   useEffect(() => {
     const fetchOptions = async () => {
       try {
-        await languageProvider.start();
-        fetchOptionsCallback('service.name', languageProvider);
-        fetchOptionsCallback('name', languageProvider);
-        setHasSyntaxLoaded(true);
+        const services = await loadOptions('serviceName');
+        setServiceOptions(services);
+        const spans = await loadOptions('spanName');
+        setSpanOptions(spans);
```

Comment:
```suggestion
        const [services, spans] = await Promise.all([loadOptions('serviceName'), loadOptions('spanName')];
        setServiceOptions(services);
        setSpanOptions(spans);
```

Just a nit, but I think it might be a good idea to pull those two async calls together.
Rest of the PR looks good to me! 👍